Fix/177 display js disabled message

Shavindra opened the pull request:
## Summary

<!-- Please reference the issue this PR addresses. -->
Addresses issue #

## Relevant technical choices

<!-- Please describe your changes. -->

## Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [ ] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

Shavindra committed:
fixed

Shavindra committed:
Added noscript message and placeholder test

Shavindra committed:
Update Requires_Javascript_Trait.php

Shavindra committed:
changed method to protected

Shavindra committed:
Merge branch 'fix/177-Display-JS-disabled-message' of https://github.com/Shavindra/site-kit-wp into fix/177-Display-JS-disabled-message

aaemnnosttv commented on tests/e2e/specs/dashboard/dashboard-noscript.js:
```diff
@@ -0,0 +1,27 @@
+	} );
+
+	beforeEach( async () => {
+		page.setJavaScriptEnabled( false );
```
This should only be necessary `beforeAll` as I don't believe it will be reset between. It should also be reset to `true` `afterAll`.

aaemnnosttv commented on includes/Core/Util/Requires_Javascript_Trait.php:
```diff
@@ -0,0 +1,49 @@
+			esc_html__( 'The Site Kit by Google plugin requires JavaScript to be enabled in your browser.', 'google-site-kit' )
+		);
+		
+		return apply_filters( 'googlesitekit_noscript_html', $no_script );
```
Please remove this filter as it is not mentioned in the AC or IB.

nicholasio replied:
that was one suggestion I gave to @Shavindra but noted it would probably be a bit overkill  here.

Would you say that if there's any need for a filter like this it would typically be mentioned in the IB correct?

aaemnnosttv replied:
Yes, usually. There should at least be a good reason for adding it. In this case, I don't think there's much value in making the output here extensible for other plugins/themes to customize.

nicholasio commented on includes/Core/Util/Requires_Javascript_Trait.php:
```diff
@@ -0,0 +1,49 @@
+ * @access private
+ * @ignore
+ */
+Trait Require_Javascript_Trait {
```
`trait`should be in lowercase.

nicholasio replied:
Also, it looks like the trait name should only be `Require_Javascript`. cc @aaemnnosttv

aaemnnosttv replied:
Most traits do have the `_Trait` suffix as part of the actual name but that may only be because they have a corresponding interface which would conflict otherwise.

The IB mentions it as a `Requires_Javascript` (not `Require_Javascript`) which is good I think.

Make sure that the filename matches the name of the trait when changing as well as any references in doc blocks, comments, annotations, etc.

nicholasio replied:
the `Migrate_Legacy_Keys` doesn't have the `_Trait` suffix that was mostly why I suggested removing it.

aaemnnosttv replied:
Yep. It may be that as we add more traits we develop more of a well-defined convention going forward but I think it's okay without the suffix for now 👍

aaemnnosttv commented on includes/Core/Util/Requires_Javascript_Trait.php:
```diff
@@ -0,0 +1,49 @@
+	/**
+	 * Display fallback message when Javascript is disabled
+	 *
+	 * @since 1.1.2
```
Use this format for all new classes, functions, and methods:
```suggestion
	 * @since n.e.x.t
```

nicholasio replied:
Is there a place to reference what the next version will be? Wondering how to know if the next one is a minor or major release.

aaemnnosttv replied:
We use `n.e.x.t` as a placeholder which is replaced when preparing the upcoming release, so you don't have to know 😄

nicholasio replied:
Ah gotcha, makes sense.

Shavindra committed:
Updated tests

Shavindra committed:
PR fixes

Shavindra committed:
added TODO

Shavindra committed:
trying to fix tests

Shavindra committed:
updated the spec

Shavindra committed:
fixed tests

nicholasio commented on includes/Core/Util/Requires_Javascript_Trait.php:
```diff
@@ -0,0 +1,49 @@
+ * @access private
+ * @ignore
+ */
+trait Requires_Javascript_Trait {
```
@Shavindra it looks like the indentation here is a bit off. There's a .editorconfig file that can help you if you have the editorconfig extension.

Shavindra committed:
Updated

Shavindra committed:
Testing with Travis CI

Shavindra committed:
fixed linting

Shavindra committed:
Merge remote-tracking branch 'refs/remotes/upstream/develop'

Conflicts:
	tests/e2e/specs/plugin-activation.test.js

Shavindra committed:
fixed

nicholasio commented on includes/Core/Util/Requires_Javascript_Trait.php:
```diff
@@ -0,0 +1,51 @@
+						<div class="mdc-layout-grid__inner">
+							<div class="mdc-layout-grid__cell mdc-layout-grid__cell--span-12">
+								<h3 class="googlesitekit-heading-3 googlesitekit-noscript__title">
+								The Site Kit by Google plugin requires JavaScript to be enabled in your browser.
```
<?php esc_html__e( '', 'google-site-kit' ); ?>

nicholasio commented on includes/Core/Admin/Screen.php:
```diff
@@ -203,6 +205,8 @@ private function render( Context $context ) {
 			return;
 		}
 
+		echo $this->get_noscript_html(); // phpcs:ignore
```
// phpcs:ignore PHPCS.rulename

Commit:
Fix 177 - Added no JS notice to activation, admin and menu

fixed tests

Added comments

fixed linting

fixed css linting

fixed

fixd CSS

Update _googlesitekit-noscript.scss
Update Requires_Javascript_Trait.php
Update Screen.php
Update Admin_Bar.php
Update Activation.php
Update Requires_Javascript_Trait.php
fixed

fixed linting

fixed linting

fixed space

fixed spacings

nicholasio commented on tests/e2e/specs/dashboard/dashboard-noscript.test.js:
```diff
@@ -0,0 +1,39 @@
+	} );
+
+	describe( 'When Javascript is disabled', () => {
+		beforeAll( async () => {
```
We discussed on the call enabling/disabling javascript within each test as it was breaking other things inside the test. I think for consistency, let's toggle JS within each test,

```
                it( 'Should not display plugin html', async () => {
                        await page.setJavaScriptEnabled( false );
			const noscript = await page.$( '#wpbody-content' );
			await expect( noscript ).not.toMatchElement( '.js-googlesitekit-plugin' );
                        await page.setJavaScriptEnabled( true );
		} );

		it( 'Should display noscript notice', async () => {
                        await page.setJavaScriptEnabled( false );
			const noscript = await page.waitForSelector( '.googlesitekit-noscript', {
				visible: true,
			} );
			await expect( noscript ).toMatchElement( '.googlesitekit-noscript__title', { text: 'The Site Kit by Google plugin requires JavaScript to be enabled in your browser.' } );
                        await page.setJavaScriptEnabled( true );
		} );
```

Shavindra replied:
This tests works out ok but I will make the change

Shavindra replied:
@nicholasio JS need to be disabled before user navigate to the the page. 

https://github.com/Shavindra/site-kit-wp/pull/3/files#diff-d24cbec8d794b4a26a34c30dd6737b77R8

Here I think it will be fine to leave it how it was

CC: @aaemnnosttv

nicholasio replied:
Gotcha, ok. @Shavindra I think we can move this to the google repo and have the official review there.

Commit:
fixed

Update plugin-activation.test.js
